### ISSUE TITLE
EIPIP Meeting 106

### ISSUE THREAD

**Issue by poojaranjan** [ethcatherders/EIPIP#340]
# Date and Time

July 03, 2024 at 17:30 UTC 

# Location

Zoom: TBA in the Discord #eip-editing channel

YouTube Recording: [EIPIP Meetings](https://www.youtube.com/playlist?list=PL4cwHXAawZxpLrRIkDlBjDUUrGgF91pQw)

# Agenda

### Edit to `Final` Proposals
- https://github.com/ethereum/EIPs/pull/8590 [Editor's decision]
- https://github.com/ethereum/EIPs/pull/8503 [Require one more editor's approval]
- https://github.com/ethereum/EIPs/pull/8607 [What's next?]

### Update to EIP-1
- https://github.com/ethereum/EIPs/pull/8634

#### Misc Issue/ requires attention?
- Resolve the EIP-7212 migration Ref: [Comment](https://github.com/ethcatherders/EIPIP/issues/340#issuecomment-2200501895)
- https://github.com/ethereum/ERCs/issues/473
- https://github.com/ethereum/ERCs/issues/243 
- https://github.com/ethereum/EIPs/pull/8716
Spam?
- https://github.com/ethereum/EIPs/issues/8718
- https://github.com/ethereum/EIPs/issues/8717 
- https://github.com/ethereum/EIPs/issues/8709
- https://github.com/ethereum/EIPs/issues/8705
- https://github.com/ethereum/ERCs/issues/297 [To be closed?]
- https://github.com/ethereum/EIPs/pull/8309 [Requested [here](https://discord.com/channels/595666850260713488/746566142700814426/1249826233760350258)]
- 
### Call for Input

Call For Input | Status | Inclination/Result | Deadline
-- | -- | -- | --
[343](https://github.com/ethcatherders/EIPIP/issues/343) | Open |  |July 28, 2024

## 2. Other discussions and updates from past meetings
- [ ]  Security issues with ERC-20 and ERCs in general. Ref: [Comment](https://github.com/ethcatherders/EIPIP/issues/340#issuecomment-2197422911)
- [ ] https://github.com/ethereum/EIPs/pull/8662
- [ ] Discuss blockers to merge a Pull Request
- [ ] Estd. timeline for merging `Draft` proposals 
  - [ ] https://github.com/ethereum/EIPs/pull/8711
  - [ ] https://github.com/ethereum/EIPs/pull/8712
  - [ ] https://github.com/ethereum/EIPs/pull/8684
  - [ ] https://github.com/ethereum/EIPs/pull/8662
  - [ ] https://github.com/ethereum/EIPs/pull/8630
  - [ ] https://github.com/ethereum/EIPs/pull/8629
  - [ ] https://github.com/ethereum/EIPs/pull/8563
  - [ ] https://github.com/ethereum/EIPs/pull/8493
  - [ ] https://github.com/ethereum/EIPs/pull/8454
  - [ ] https://github.com/ethereum/EIPs/pull/8452
  - [ ] https://github.com/ethereum/EIPs/pull/8372
  - [ ] https://github.com/ethereum/EIPs/pull/8267
  - [ ] https://github.com/ethereum/EIPs/pull/8261
  - [ ] https://github.com/ethereum/EIPs/pull/8190

## 3. EIPs Insight - Monthly EIPs status reporting.

Repo | Issues | PRs | Total
-- | -- | -- | --
EIP |  11 | 76  | 87
ERC | 4  | 71  |75
RIP |  1 | 11 | 12

([Dashboard](https://hackmd.io/@poojaranjan/EthereumImprovementProposalsInsight/https%3A%2F%2Fhackmd.io%2F%40poojaranjan%2FEIPsInsightJune2024#Dashboard))

- [EIPsInsight.com](https://eipsinsight.com/insight/2024/6)
- [Hackmd](https://hackmd.io/@poojaranjan/EthereumImprovementProposalsInsight)

## 4. [EIP Editing Office Hour](https://hackmd.io/@poojaranjan/EIPEditingOfficeHour)
- Agenda [39](https://github.com/ethcatherders/EIPIP/issues/341) Video [38](https://www.youtube.com/watch?v=cfhXAAKDYoM&list=PL4cwHXAawZxqnDHxOyuwMpyt5s8F8gdmO&index=1&t=123s&pp=gAQBiAQB)
  - 13/13 PRs Reviewed
 

## 5. Review action items from earlier meetings
- [Summary meeting 105](https://github.com/ethcatherders/EIPIP/issues/337#issuecomment-2151133185)
Action items from the last meeting
- Make a _Call For Input_ to collect thoughts from other Editors for https://github.com/ethereum/EIPs/pull/8390.

## Next Meeting date & time 

July 17, 2024 at 17:30 UTC

**Comment by Dexaran** [comment]
I would like to attend the EIP editors meeting to discuss security issues with ERC-20 and ERCs in general. I was raising this concern some time ago, there was some brainstorming done (https://ethereum-magicians.org/t/modification-of-eip-process-to-account-for-security-treatments/16265/26 and this https://hackmd.io/@SamWilsn/HJL1ydMVp) but there was no outcome and it seems this topic is not moving.

So I would like to:
- Highlight the existing challenges of the EIP process from a security perspective.
- Hear EIP editors feedbacks
- Discuss future steps that could be taken in order to solve any of the mentioned problems
- Discuss the state of ERC-20 "Known issues". Over a year ago we discussed the possibility of adding a warning and my proposal was voted out by @gcolvin if I'm correct. A year passed, the issue is not solved, the amount of lost funds increased. May be it requires different approach.

**Comment by frangio** [comment]
Consider discussing how to resolve the EIP-7212 migration as it is still up and appears valid even though the current spec is RIP-7212. Among other things the precompile address is different from the one in RIP-7212 and i'm concerned someone might mistake it for the true spec.

- https://github.com/ethereum/EIPs/pull/8101

**Comment by Dexaran** [comment]
We had a discussion regarding the EIP editors call where I mentioned a number of comments and statements that were made during 2015, 2019 and 2023 as well as few other EIPs and ERCs that were created. It will take some time to pack it all together for me and I have a conference tomorrow that I have to prepare to. 

I will edit this comment and replace this text with the recap of my proposals in a written form as well as all the associated links (hopefully tomorrow, or may be on 5th July if I'll be too overloaded).

**Comment by poojaranjan** [comment]
## Summary
### 1. Discuss Open Issues/PRs, and other topics
https://github.com/ethereum/EIPs/pull/8590 
- Waiting on @lightclient 's approval. 

https://github.com/ethereum/EIPs/pull/8634
- Was waiting on 3rd editor's approval. @xinbenlv approved it and is merged.

https://github.com/ethereum/ERCs/issues/473
- Keep the issue open. It could be a good Issue to be worked on by someone willing to make open-source contribution. 

https://github.com/ethereum/ERCs/issues/243
- Keep the issue open. Need to be worked on.

Spam Issues - Closed

### 2. Other discussions and updates from past meetings
 #### Security issues with ERC-20 and ERCs in general. Ref: 
https://github.com/ethcatherders/EIPIP/issues/340#issuecomment-2197422911
- @Dexaran provided an overview of the issue and proposed solutions. Ref:[Comment](https://github.com/ethcatherders/EIPIP/issues/340#issuecomment-2207486824)
- To follow the conversation, follow the [recording](https://www.youtube.com/watch?v=PKkJNqcozhw&t=710s).

#### Discuss blockers to merge a Pull Request

@poojaranjan shared main reasons identified for open PRs
* **Waiting for Editor** Consensus: Many pull requests are pending agreement among the editors.
* **Waiting for Author Review**: Some proposals require the original authors to review and approve changes.
* **Draft Status**: Some EIPs are still in draft form and need further development before they can be merged.
* **CI Tests Pending**: Several proposals are waiting for Continuous Integration (CI) tests to pass.

- Client devs are looking for 'Draft` PRs to be merged sooner to move the discussion to the FEM page rather than on the PR itself.
- @SamWilsn agreed to the idea of merging `Draft` PRs 
- @xinbenlv also recommended the same for ERC proposals

 ##### Estd. timeline for merging Draft proposals
 - @SamWilsn [demoed a tool](https://github.com/gaudren/EIP-Board/tree/events) that can help identify the open PRs to the editor in order of last engaged with the author.
 - This will help get an estimated timeline by which the PR will be reviewed by an editor. The idea is to add this tool to the eips.ethereum.org.
 - Editors will look into open PRs added to the agenda and merge them.

Also, the group discussed merging the `Draft` PR with a minimum requirement. Most of the editing reviews will be done when the EIP is moved to `Review` status.

**Comment by poojaranjan** [comment]
Closing in favor of #346

**Comment by Dexaran** [comment]
@poojaranjan I've updated the comment https://github.com/ethcatherders/EIPIP/issues/340#issuecomment-2207486824

Links were provided, my proposals are stated.

@SamWilsn 

"Do we need to add issues like sending to wrong address to the text of each ERC describing a token that can be sent to wrong address?"

Sending to wrong address is a protocol-level issue. It is related to what is considered a valid "account" in Ethereum. Considering any pattern of 42 hexadecimal symbols as a valid transaction recipient is not the best idea. In theory the ENS was designed to solve it but it failed to gain sufficient adoption.

There are alternative implementations. For example in EOS there is a built-in naming service and this is how a EOS account looks like https://www.bloks.io/account/dexaran.x https://www.bloks.io/account/dexaraniiznx

This is how a contract looks like in EOS https://www.bloks.io/account/tethertether?loadContract=true&tab=Tables&account=tethertether&scope=tethertether&limit=100

And if you will try to access an EOS account that doesn't exist (i.e. it is not assigned to any public key) then you will not be able to do this https://www.bloks.io/account/dexarannn404

Similarly, if you would try to send funds to a "non-existing account" in EOS - the transaction will fail.

Is sending to wrong address a problem? - Yes, it is. In my opinion it can be classified as a security problem because it threatens the safety of users funds. But this is a different problem than the one I was reporting earlier. And it is a protocol-level problem. 

Is it worth adding a protocol-level problem to every text of every EIP which is affected? - In my opinion no. If there is a EIP that describes how addresses are derived from a private key in Ethereum then it would be the right place to mention this problem. 

If you would like to completely remove the burden of making this types of decisions from EIP editors then I would support it either. Anyways it's better to have a protocol-level issue described in every token ERC and ERC-20 issue described in ERC-20 text than not to have any issues described at all.

I would propose to implement an anti-spam threshold i.e. "If there are more than X amount of $ lost because of a certain problem then it must be added to the security considerations section of affected EIPs/ERCs" so that to avoid describing spam issues that pose no real threat. For example you can deliver Ether to a smart-contract via `SELFDESTRUCT` bypassing the fallback function invocation but this issue caused about $0 losses to the users of the ecosystem during the past 7 years unlike the ERC-20 issue which caused about $80M-$230M. 

P.S. The same applies to counterfactual contracts. Before the contract is created it mustn't be considered an entity capable of receiving a financial transaction. This bypasses fallback function invocation and must be considered an anti-pattern in general. Ether can also be lost this way and it's a similar issue of "account generation".
